cmake/extensions.cmake

```diff
@@ -826,7 +826,6 @@ function(board_finalize_runner_args runner)
 endfunction()
 
 function(board_set_rimage_target target)
-  zephyr_check_cache(RIMAGE_TARGET)
```

Comment:
If you want to ensure this value doesn't change, then you can just move it below.

Like this:
```
function(board_set_rimage_target target)
  set(RIMAGE_TARGET ${target} CACHE STRING "rimage target")
  zephyr_check_cache(RIMAGE_TARGET)
endfunction()
```

This will set the value on first invocation, and then on subsequent invocation ensure it doesn't change.

That is:
```
$ cmake .
Including boilerplate (Zephyr base (cached)): /projects/github/ncs/zephyr/cmake/app/boilerplate.cmake
....
-- Build files have been written to: /projects/github/ncs/zephyr/build
```

but doing:
```
$ cmake . -DRIMAGE_TARGET=test
Including boilerplate (Zephyr base (cached)): /projects/github/ncs/zephyr/cmake/app/boilerplate.cmake
....
CMake Warning at /projects/github/ncs/zephyr/cmake/extensions.cmake:2393 (message):
  The build directory must be cleaned pristinely when changing rimage target,

  Current value="cnl", Ignored value="test"
Call Stack (most recent call first):
...
```

Reply:
Aha, thank you @tejlmand !